internal/advertisement-operator/broadcaster.go

```diff
@@ -325,6 +326,29 @@ func (b *AdvertisementBroadcaster) SendAdvertisementToForeignCluster(advToCreate
 	return adv, nil
 }
 
+func (b *AdvertisementBroadcaster) NotifyAdvertisementDeletion() error {
+	advName := "advertisement-" + b.HomeClusterId
+	obj, err := b.RemoteClient.Resource("advertisements").Get(advName, metav1.GetOptions{})
+	if err != nil {
+		klog.Error("Advertisement " + advName + " doesn't exist on foreign cluster " + b.ForeignClusterId)
+	} else {
+		// update the status of adv to inform the vk it is going to be deleted
+		adv := obj.(*protocolv1.Advertisement)
+		adv.Status.AdvertisementStatus = AdvertisementDeleting
+		_, err = b.RemoteClient.Resource("advertisements").UpdateStatus(adv.Name, adv, metav1.UpdateOptions{})
+		if err != nil {
+			klog.Error("Unable to update Advertisement " + adv.Name)
+			return err
+		}
+	}
+	// delete the peering request to delete the broadcaster
+	if err := b.DiscoveryClient.Resource("peeringrequests").Delete(b.PeeringRequestName, metav1.DeleteOptions{}); err != nil {
```

Comment:
Await for advertisement to be deleted to avoid PeeringRequest recreation

If ForeignCluster operator has an update between PR delete and Adv delete, it is aware of that process and it will create new PR

Reply:
thanks!